### ISSUE TITLE
Single region SNR calculation always expects uncertainties

### ISSUE THREAD

**Issue by robelgeda** [astropy/specutils#282]
[This line](https://github.com/astropy/specutils/blob/1014a877af178a255ca2b0ceae9e6ab96a451433/specutils/analysis/snr.py#L77) assumes that the `Spectrum1D.uncertainty` is not `None`.

**Comment by nmearl** [comment]
cc: @brechmos-stsci

**Comment by brechmos-stsci** [comment]
The decision in the implementation was that SNR must have uncertainties defined. Previous to that I was thinking of adding a third parameter (default None) that is a Region object called `noise_region` over-which the "noise" would be calculated *if* `noise_region` was not None.

In discussions with @eteq (not to throw you under the bus) my understanding was that the SNR should be calculated from the uncertainty defined in the Spectrum1D object.   

Oh, and as I think about it now, I think the idea was that if someone did not know the uncertainty then they should calculate the noise measure (e.g., RMS) in some manner and add that as the uncertainty in the Spectrum1D object's uncertainty attribute.

tl;dr  requiring uncertainty was a design decision

Now, the fact that it throws an exception in that manner was a bit of a design decision too. I was thinking of wrapping it in a try/except and then print the error message and raise an exception, but it seemed like leaving the line like that was, well, sufficient. I can definitely add the try/except in there.

**Comment by eteq** [comment]
Oh, this *does* raise the interesting point that perhaps this should be mentioned somewhere in the narrative docs.  That is, maybe we should add a page under [Basic Analysis](https://specutils.readthedocs.io/en/latest/basic_analysis.html) that shows a simple example of estimating the uncertainty with `noise_region_uncertainty` and then computing the SNR?

If that sounds good, this issue could be kept open and just renamed that we should *document* this behavior.

**Comment by robelgeda** [comment]
In that case something like this would be good to add:
```
if calc_spectrum.uncertainty is None:
    raise Exception("Spectrum1D.uncertainty is undefined.")
```
